ts-rs/tests/unit.rs

```diff
@@ -5,7 +5,7 @@ use ts_rs::TS;
 struct Unit;
 
 // serde_json serializes this to `{}`.
-// The TS type best describing an empty object is `Record<string, never>`.
+// The TS type best describing an empty object is `{ }`.
```

Comment:
I'm curious about this here!
I actually think `Record<string, never>`, while being clunky, is way more appropriate here.
The type `{ }` pretty much allows anything except `null` and `undefined`:
```ts
type Unit = {};

const a: Unit = 123;
const b: Unit = "hey!";
const c: Unit = [1, 2, 3];
const d: Unit = new Map();
const e: Unit = { a: [], b: "hello world" };
const f: Unit = {};
// const g: Unit = null; ERROR
// const h: Unit = undefined; ERROR
```

`Record<string, never>`, on the other hand, actually represents an empty object:
```ts
type Unit = Record<string, never>;

// const a: Unit = 123; ERROR
// const b: Unit = "hey!"; ERROR
// const c: Unit = [1, 2, 3]; ERROR
// const d: Unit = new Map(); ERROR
// const e: Unit = { a: [], b: "hello world" }; ERROR
const f: Unit = {};
// const g: Unit = null; ERROR
// const h: Unit = undefined; ERROR
```

My TS knowledge is wayyyy to basic to explain why `{}` behaves like this, but I guess it does.

Reply:
> The type `{ }` pretty much allows anything except `null` and `undefined`

This is correct, I just removed it because there seems to be some controversy around the topic [as this thread shows](https://github.com/microsoft/TypeScript/issues/47486#issuecomment-1015542476)

I also did some testing and, although you can't create a `Record<string, never>`, if given one you can read from it and get a `never`, but no error
```ts
declare const test: Record<string, never>
const a = test.foo // no error
   // ^? never
```

Reply:
Ah, I see. That sucks.
How about `Record<never, never>`?

Reply:
As for why `{}` behaves the way it does, take the following code:
```ts
interface Point {
  x: number
  y: number
}

function doSomething(p: Point) {
  return
}

const point = { x: 0, y: 0, z: 0 }
doSomething(point) // No Error
```

In general TS types mean "Object must contain **at least** these properties", not "Object must contain **exactly** this properties", and any type (other than `null` and `undefined`) contain at least zero properties (e.g.: pretty much everything has `.toString`)

Reply:
As for `Record<string, never>` it's behaves like "any string may be a key to this object, but no value may be assigned to any of these keys"

Reply:
> In general TS types mean "Object must contain at least these properties"

That's how it works in e.g Java, but I don't think that's the entire trough. For example, this here does **not** work in TS. It would work in Java, if `{ a: number, b: number }` was considered a subtype of `{ a: number }`.

```ts
let y: { a: number } = { a: 1, b: 2 };
```

Reply:
It seems like TypeScript is more strict about it when it has an object literal. For example, in the code I presented before, the following is invalid:
```ts
interface Point {
  x: number
  y: number
}

function doSomething(p: Point) {
  return
}

doSomething({ x: 0, y: 0, z: 0 })// Error object literal may only specify known properties, and z does not exist in type Point
```

No idea why that is though

Reply:
> Ah, I see. That sucks.
> How about `Record<never, never>`?

This thread got so big I completely missed this comment

`Record<never, never>` gets simplified down to `{}`:
![image](https://github.com/Aleph-Alpha/ts-rs/assets/131818645/686c6008-803a-4925-8d24-c71253600eff)

Reply:
I think there is no way to express that a type can only be assigned an empty object (like `Record<string, never>`) and at the same time get an error reading from it (like `{}`)